README.md

```diff
@@ -1,70 +1,118 @@
+Moreover, this encoding does not reduce the security of the system. 
+Even if there is an algorithm to solve ECDLP for the xonly encoding, then the full encoding will also be automatically broken since the `X` coordinate has two possible `Y` coordinates as negative and positive.
 
-### Progress
+## Adapting MuSig2 to PIB-430
 
-- [x] Update the function and parameter namings.
-- [x] Prevent reuse of r values.
-- [x] Create musig2 context to simplify the API.
-- [ ] Tests (Valgrind).
+[MuSig2](https://eprint.iacr.org/2020/1261.pdf) 
+is a two-round multi-signature scheme that outputs an ordinary Schnorr signature.
+
+$$ G \cdot s = R + X \cdot c. $$
 
+The aggregated public key $X$ is generated with the public keys of all signers:
 
+$$ X = \Sigma_{i = 1}^{N} (X_i \cdot a_i) $$
 
+where $a_i$ is the key aggregate coefficient.
+The commitment $R$ is also generated with the commitments of signers and the nonce $b$.
 
+$$ R = \Sigma_{i = 1}^{V}(R_j^{b^{j-1}}). $$
 
+In order to verify MuSig2 by using the function `secp256k1_schnorrsig_verify`, the signature must be obtained by $X$ and $R$ with even `Y` coordinates.
+However, the aggregation of public keys and the commitments may not result points with even `Y` coordinates. 
+Therefore, signers do the following in the single signature generation step:
 
 
+- **Case 1** ( $X$ has odd, $R$ has even `Y` coordinate): In this case, signers negate $c$. 
+Negating $c$ is equivalent to negating $X$, since
 
+$$ G \cdot s = R + (-X) \cdot (-c) $$
 
+- **Case 2** ( $X$ has even, $R$ has odd `Y` coordinate): In this case, 
+signers negate every element in the list $b^0, b^1, \ldots, b^{V-1}$, so $R$ is also negated.
 
+- **Case 3** (Both have odd `Y` coordinates): Signers negates their partial signature, so that the aggregated signature will be negated:
 
+$$ G \cdot (-s) = (-R) + (-X) \cdot c. $$
 
+### AOMDL
+The security of MuSig2 depends on the hardness of *algebraic one more discrete logarithm (AOMDL)* problem. 
+In the standard one more discrete logarithm problem, an adversary is given the public parameters $(E(F_p), G, p)$ and wins if it can solve the discrete logarithms $(x_1, \ldots, x_{q+1})$ of $q+1$ challenge group elements $(X_1, \ldots, X_{q+1})$ by only making at most $q$ queries to $D_{LOG}$ oracle.
+The algebraic version of this problem requires to include an algebraic representation $(\alpha, (\beta_i)_{1 \leq i \leq c})$ 
+of the challenge $X$ such that:
 
+$$ X = G \cdot \alpha + \Sigma_{i = 1}^{c}(X_i \cdot \beta_i). $$
 
+The question is that whether using xonly encoding for the public keys reduce the security of the scheme.
+If there exists a polynomial algorithm to solve AOMDL for xonly encoding of a curve point, then it basically implies that AOMDL is broken for full-size encoding as well, since every `X` coordinate has two possible `Y` coordinate as odd or even. 
```

Comment:
Same as above, we probably want a more detailed argumentation of this. My question is, is the AOMDL affected in any way by the changes we make? Which part of the protocol depends on this? probably the exponents `b_i`, right? 🤔